agent/agentssh/forward.go

```diff
@@ -209,3 +233,14 @@ func directStreamLocalHandler(_ *ssh.Server, _ *gossh.ServerConn, newChan gossh.
 
 	Bicopy(ctx, ch, dconn)
 }
+
+// unlink removes files only.
+func unlink(path string) error {
+	// From os/file_posix.go:
+	for {
+		err := syscall.Unlink(path)
```

Comment:
I don't understand why we're using syscall directly instead of the higher level `os` functions.  Is it just because we only want to unlink files and `os.Remove()` removes directories?

If we are using syscalls, do we need to include build flags so we don't try this on Windows and return a sensible error?

Reply:
I wanted to match OpenSSH behavior (we only remove files, not folders). Without doing stat (which also is racy if you think about it), there's no other way to achieve the same (AFAIK).

Since this compiles fine on Windows, I thought it wouldn't be a problem, but I can try to verify. AFAIK this implementation is not supported on Windows, though.

Reply:
Fine to match OpenSSH behavior --- maybe a comment to that effect would be helpful.

And, not for this PR, but I know that we _do_ have customers that target Windows workspaces, and we should verify we get some sensible error if you try to remote-forward a Unix socket from one.

Reply:
There is a comment to this effect 😄 https://github.com/coder/coder/pull/11631/files#diff-7ac7e219a582bce4b340900ae8681925a65e509602ee869a2863ea2268b64d60R105

> And, not for this PR, but I know that we do have customers that target Windows workspaces, and we should verify we get some sensible error if you try to remote-forward a Unix socket from one.

For sure 👍🏻, I'll make an issue for it.

Reply:
https://github.com/coder/coder/issues/11649

Reply:
@spikecurtis FYI, just confirmed `syscall.Unlink` works on Windows too, same behavior as Unix.